src/main/java/ch/njol/skript/expressions/ExprStructureSectionInfo.java

```diff
@@ -0,0 +1,185 @@
+@RequiredPlugins("Minecraft 1.17.1+")
+@Since("INSERT VERSION")
+public class ExprStructureSectionInfo extends SimpleExpression<Object> {
+
+	static {
+		if (Skript.classExists("org.bukkit.structure.Structure")) {
+			Skript.registerExpression(ExprStructureSectionInfo.class, Object.class, ExpressionType.SIMPLE,
+					"includes entities", "rotation", "integrity", "pallet [index]", "mirror"
+			);
+		}
+	}
+
+	private enum Setting {
+		INCLUDES_ENTITIES(Boolean.class),
+		ROTATION(Rotation.class),
+		INTEGRITY(Float.class),
+		PALLET(Integer.class),
+		MIRROR(Mirror.class);
+
+		private final Class<? extends Object> returnType;
+
+		Setting(Class<? extends Object> returnType) {
+			this.returnType = returnType;
+		}
+
+		public Class<? extends Object> getReturnType() {
+			return returnType;
+		}
+	}
+
+	private Setting setting;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentSection(EffSecStructurePlace.class)) {
+			Skript.error(parseResult.expr + " can only be used in a structure place section!");
+			return false;
+		}
+		setting = Setting.values()[matchedPattern];
+		return true;
+	}
+
+	@Override
+	protected Object[] get(Event event) {
+		if (!(event instanceof StructurePlaceEvent))
+			return new Object[0];
+		StructurePlaceEvent details = (StructurePlaceEvent) event;
+		switch (setting) {
+			case INCLUDES_ENTITIES:
+				return CollectionUtils.array(details.includeEntities());
+			case INTEGRITY:
+				return CollectionUtils.array(details.includeEntities());
+			case PALLET:
+				return CollectionUtils.array(details.includeEntities());
+			case MIRROR:
+				return CollectionUtils.array(details.getMirror());
+			case ROTATION:
+				return CollectionUtils.array(details.getRotation());
+		}
+		return new Object[0];
+	}
+
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode != ChangeMode.SET)
+			return null;
+		if (setting.getReturnType().isAssignableFrom(Number.class))
+			return CollectionUtils.array(Number.class);
+		return CollectionUtils.array(setting.getReturnType());
+	}
+
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null)
+			return;
+		if (!(event instanceof StructurePlaceEvent))
+			return;
+		StructurePlaceEvent details = (StructurePlaceEvent) event;
+		switch (setting) {
+			case INCLUDES_ENTITIES:
+				details.setIncludesEntities((boolean) delta[0]);
+				break;
+			case INTEGRITY:
+				float integrity = ((Number) delta[0]).floatValue();
+				details.setIntegrity(Math.min(1, Math.max(0, integrity)));
+				break;
+			case PALLET:
+				int pallet = ((Number) delta[0]).intValue();
+				int max = details.getStructure().getPaletteCount();
+				details.setPallet(Math.min(max, Math.max(-1, pallet)));
+				break;
+			case MIRROR:
+				details.setMirror((Mirror) delta[0]);
+				break;
+			case ROTATION:
+				details.setRotation((StructureRotation) delta[0]);
+				break;
+		}
```

Comment:
Instead of a switch here, include a consumer in the enum so it's all in one place.